rtv/oauth.py

```diff
@@ -99,15 +110,25 @@ def authorize(self):
         self.open_config(update=True)
         # If no previous OAuth data found, starting from scratch
         if not self.config.has_section('oauth') or not self.config.has_option('oauth', 'refresh_token'):
+            # Start HTTP server and listen on port 65000
+            self.callback_app.listen(65000)
+            
             # Generate a random UUID
             hex_uuid = uuid.uuid4().hex
 
             permission_ask_page_link = self.reddit.get_authorize_url(str(hex_uuid),
                 scope=self.scope, refreshable=True)
 
-            with self.loader(message='Waiting for authorization'):
-                webbrowser.open(permission_ask_page_link)
+            if self.compact:
+                show_notification(self.stdscr, ['Opening ' + os.environ.get('BROWSER')])
+                curses.endwin()
+                webbrowser.open_new_tab(permission_ask_page_link)
                 ioloop.IOLoop.current().start()
```

Comment:
Have you tested this out? I still get stuck waiting for a reply from ``127.0.0.1:65000``. I assume this is because the iloop isn't started until after the webbrowser closes, but the webbrowser won't redirect unless the iloop has already started. I haven't used tornado before but it looks like there's some anync stuff built-in (``iloop.run_sync()``) that may be useful.